### PR TITLE
feat/thin-11: adds API for creation/retrieval of pins

### PR DESCRIPTION
**Pull request by abdullahtariq91**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,69 @@
...
+We use `yarn` to install the packages. Do
+```
+ yarn install
+```
+in the project's directory. After installing you need to set up environment variables. In Unix based systems you can do:
+```
+ export TB_PIN_DB_STRING="Mongo db string here"
+```
+
+> Contact the collaborators to get remote db string and node string.
```
Comment (hamxabaig) [inline README.md:15]: @abdullahtariq91 Add this remote db string to nuclino's `Database` page. We document there everything

File: app/models/pin.model.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Schema, model, Document } from 'mongoose';
+
+const PinsSchema = new Schema({
+  code: {
+    type: String,
+    required: true
+  },
+  phone: {
+    type: String,
+    required: true
```
Comment (hamxabaig) [inline app/models/pin.model.ts]: @abdullahtariq91 it should be unique?

File: app/modules/pin/pin.controller.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import { IRequest, IResponse } from '../../interfaces/utils/IServer';
+import Pin from '../../models/pin.model';
+import * as restify from 'restify';
+
+export default class PeriodicalCronListener implements IController {
+  public async post(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const pin = req.body;
+
+      const saved = await Pin.create({ code: pin.code, phone: pin.phone });
```
Comment (hamxabaig) [inline app/modules/pin/pin.controller.ts]: Check if the phone is unique and if its not, return an error to the user.

File: app/modules/pin/pin.controller.ts [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+  public async get(req: IRequest, res: IResponse, next: restify.Next) {
+    try {
+      const pin = await Pin.findOne({ phone: req.params.phone });
+
+      if (!pin) {
+        return res.send(new restify.NotFoundError('Pin not found for given phone'));
+      }
+
+      return res.send(pin);
```
Comment (hamxabaig) [inline app/modules/pin/pin.controller.ts]: I think after retrieving the pin from db, we'll send it to users phone using twilio?

File: app/modules/pin/pin.route.ts [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+class PinRoute implements IRoute {
+  public basePath = '/pins';
+  public controller = new PinController();
+
+  public getServerRoutes(): IRouteConfig[] {
+    return [
+      {
+        method: HttpMethods.POST,
+        auth: AuthStrategies.PUBLIC,
```
Comment (hamxabaig) [inline app/modules/pin/pin.route.ts]: @abdullahtariq91 ask Nestor if it should be OAUTH or JWT auth. As maybe it doesn't makes sense to leave it public?

**Review by hamxabaig — COMMENTED** [review]
@abdullahtariq91 few questions

File: README.md (hunk 1 above)
Comment (abdullahtariq91) [reply README.md:15]: Sure. 👍

File: app/modules/pin/pin.controller.ts (hunk 3 above)
Comment (abdullahtariq91) [reply app/modules/pin/pin.controller.ts]: Will do.

File: app/modules/pin/pin.controller.ts (hunk 4 above)
Comment (abdullahtariq91) [reply app/modules/pin/pin.controller.ts]: Have asked Nestor for clarification. Wasn't mentioned earlier.

**Pull request merged**